modules/thunder_news_article/thunder_news_article.install

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * @file
+ * The install file.
+ */
+
+use Drupal\user\Entity\Role;
+use Drupal\Core\Config\FileStorage;
+
+/**
+ * Implements hook_install().
+ *
+ * If workflow module is enabled, we try to add the news article to the
+ * editorial workflow.
+ */
+function thunder_news_article_install(): void {
+  try {
+
+    // Copy existing permissions for articles to news article.
+    $roles = Role::loadMultiple();
+    foreach ($roles as $role) {
+      if ($role->hasPermission('create article content')) {
+        $role->grantPermission('create news_article content');
+      }
+      if ($role->hasPermission('edit any article content')) {
+        $role->grantPermission('edit any news_article content');
+      }
+      if ($role->hasPermission('edit own article content')) {
+        $role->grantPermission('edit own news_article content');
+      }
+      if ($role->hasPermission('delete any article content')) {
+        $role->grantPermission('delete any news_article content');
+      }
+      if ($role->hasPermission('delete own article content')) {
+        $role->grantPermission('delete own news_article content');
+      }
+
+      if ($role->hasPermission('delete article revisions')) {
+        $role->grantPermission('delete news_article revisions');
+      }
+      if ($role->hasPermission('revert article revisions')) {
+        $role->grantPermission('revert news_article revisions');
+      }
+      if ($role->hasPermission('view article revisions')) {
+        $role->grantPermission('view news_article revisions');
+      }
+      if ($role->hasPermission('access_unpublished node article')) {
+        $role->grantPermission('access_unpublished node news_article');
+      }
+      $role->save();
+    }
```

Comment:
You can create an array `['has_permission' => 'grant_permission']` out of this

Reply:
doh